pkg/startup/startup_test.go

```diff
@@ -1111,3 +1112,36 @@ var _ = Describe("UTs for monitor-addresses option", func() {
 		Expect(getMonitorPollInterval()).To(Equal(30 * time.Minute))
 	})
 })
+
+var _ = Describe("UTs for env IP", func() {
+	Context("env IP is defined", func() {
+		It("get the local cidr", func() {
+
+			cmd := exec.Command("ip", "addr", "add", "1.2.3.4/24", "dev", "lo")
+			if err := cmd.Run(); err != nil {
+				Skip("ip addr add failed:" + err.Error())
+			}
+
+			ipv4Env := "1.2.3.4"
+			ipv4CIDROrIP, _ := getLocalCIDR(ipv4Env, 4)
+			Expect(ipv4CIDROrIP).To(Equal("1.2.3.4/24"))
```

Comment:
If this assertion fails then I think we'll fail to clean up the IP address that we added in the first step.

I think we probably want to include the cleanup in an `AfterEach()` or `defer()` so that it is always executed.

Alternatively, and likely preferably, we could mock out the `autodetection.GetInterfaces()` function so that we don't even need to interact with the host. Then we could have our fake "GetInterfaces()" return whatever interfaces we'd like for the tests without modifying the host's networking.

Reply:
Amazing!
 I will modify and retest

Reply:
fixed.
Constructing fake` autodetection.GetInterfaces`  is not feasible for built-in functions, so I use `defer()` . 
Do you think this test case needs to be deleted?

Reply:
I was imagining we could do something like changing the arguments to `getLocalCIDR`.

e.g.,

```
func getLocalCIDR(ip string, version in, getInterfaces func() ([]Interface, error)) (string, error) {}
```

Then, we can pass `autodetection.GetInterfaces` in the real code, and our own implementation in the test code.

Reply:
I understand what you mean. 
Yesterday I was also entangled in whether to pass in the interface as a parameter. This is why it took so long for me to update the submission.

Ok i will make changes again

Reply:
fixed,thanks.
ptal @caseydavenport